File: Software/Backend/backend/src/main/java/lk/slpa/mpma/backend/model/Lecture.java
package lk.slpa.mpma.backend.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import java.util.Date;
import lombok.Data;

@Entity
@Table(name = "lecture")
@Inheritance(strategy = InheritanceType.JOINED)
@Data
public class Lecture {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "lecture_id", nullable = false)
  private Long lectureId;

  // Replace courseId with proper relationship
  @JsonBackReference
  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "course_id", nullable = false)
  private Course course;

  // Replace personId with proper relationship
  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "lecturer_id", nullable = false)
  private Lecturer lecturer;

  @Column(nullable = false)
  private String title;

  @Column(nullable = false)
  private Date startDate;

  @Column(nullable = false)
  private String location;

  // Change duration from Date to Integer (representing minutes)
  @Column() private Integer durationMinutes;

  @Column(nullable = false)
  private String description;
}

File: Frontend/mpma/src/types/Lecture.ts
export interface Lecture {
  lectureId: number;
  courseId: number;
  lecturerId: number;
  lecturerName: string;
  title: string;
  startDate: string; // ISO 8601 date string
  durationMinutes: number;
  location: string;
  description: string;
}

export interface LectureCreate {
  title: string;
  description: string;
  startDate: string; // ISO 8601 date string
  durationMinutes: number;
  location: string;
  // courseId and lecturerId/username will be handled by the backend/service
}
